spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -170,31 +182,86 @@ public interface OrderRepository extends CrudRepository<Order, Long> {
 }
 ----
 
-For queries by method name, the resolution algorithm starts by interpreting the whole part (AddressZipCode) as the property and checks the domain class for a property with that name (uncapitalized). If the algorithm succeeds, it uses that property.
+The resolution algorithm for identifying properties in query methods by method name, with manual traversal points, is defined as follows:
+
+1. *Method Name Parsing*:: The query method's name is parsed to identify the property or properties being referenced. Method names in query methods typically follow a pattern of "findBy[Property]", where "[Property]" represents the name of the property you want to query by.
+
+2. *Property Extraction*:: The property name is extracted from the method name by removing the "findBy" prefix. For example, in the query method `findByAddressZipCode`, the property name extracted is `AddressZipCode`.
+
+3. *Property Name Capitalization*:: The extracted property name is treated as is, with its original capitalization. For example, if the property name is `AddressZipCode`, it remains in camel case.
+
+4. *Manual Traversal Points*:: To resolve ambiguity or to specify traversal through nested properties, underscores (`_`) can be used within the method name. Each underscore represents a traversal point to access nested properties. For example, `findByAddress_ZipCode` explicitly indicates traversal to the `Address` object's `ZipCode` property.
+
+5. *Domain Class Property Lookup*:: The framework checks the domain class associated with the repository for a property with the same name as the extracted property name (uncapitalized). If the domain class has a property named `addressZipCode`, this is considered a successful resolution.
+
+6. *Nested Property Handling*:: If the extracted property name includes underscores (`_`) indicating nested traversal, the framework follows the specified path to resolve the property.
```

Comment:
Should we state that you should not define entity attributes with underscore in them because it could lead to collisisons?

Reply:
Yes, I could include that we are expecting the fields follow the Java convention name to the fields.